corehq/apps/case_importer/tracking/filestorage.py

```diff
@@ -17,21 +18,27 @@ def __init__(self, bucket, meta_model):
             must contain columns identifier, filename, length
         """
 
-        self._bucket = bucket
+        self._bucket_prefix = bucket
         self._db = get_blob_db()
         self._meta_model = meta_model
 
+    def _get_bucket(self, padding):
+        return '{}/{}'.format(self._bucket_prefix, padding)
+
     def write_file(self, f, filename):
-        blob_info = self._db.put(f, bucket=self._bucket)
-        self._meta_model(identifier=blob_info.identifier, filename=filename,
+        padding = random_url_id(16)
+        blob_info = self._db.put(f, bucket=self._get_bucket(padding))
+        identifier = '{}/{}'.format(padding, blob_info.identifier)
+        self._meta_model(identifier=identifier, filename=filename,
```

Comment:
Sorry this is just indirection. `self._meta_model` is set to `CaseUploadFileMeta`.